### PR TITLE
ui#1576 - API keys - front end

### PR DESCRIPTION
**Pull request by qijieye**

### CODE REVIEW

File: Client/Api-keys/index.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+			`${this.folder}/${identifier}`
+		)
+		return result
+	}
+	async getAllApiKeys(): Promise<
+		model.ApiKeyResponse[] | (model.ErrorResponse & { status: 400 | 403 | 404 | 500 | 503 }) | undefined
+	> {
+		const result = await this.connection.get<model.ApiKeyResponse[], 400 | 403 | 404 | 500>(this.folder)
+		return result
+	}
```
Comment (lucy-meachem) [inline Client/Api-keys/index.ts]: pagination needed?

File: model/ApiKeyResponse.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { DateTime } from "isoly"
+
+export interface ApiKeyResponse {
+	identifier: string
+	displayName: string
+	rolesets?: string[]
+	roles?: string[]
+	lastKeyUsage: DateTime
+	active?: boolean
```
Comment (lucy-meachem) [inline model/ApiKeyResponse.ts]: always present

File: model/SearchRolesetsRequest.ts [hunk 3]
```diff
@@ -5,4 +5,6 @@ export interface SearchRolesetsRequest {
 	containsRoles?: string[]
 	doesNotContainRoles?: string[]
 	includeInternal: "INCLUDE" | "EXCLUDE" | "ONLY"
+	includeShared: "INCLUDE" | "EXCLUDE" | "ONLY"
+	appliesTo?: "USER" | "API_KEY"
```
Comment (lucy-meachem) [inline model/SearchRolesetsRequest.ts:9]: did you add appliesTo on the roleset response? needed?

File: Client/Api-keys/index.ts (hunk 1 above)
Comment (qijieye) [reply Client/Api-keys/index.ts]: stick 500 size in the request

**Review by ChriJ — APPROVED** [review]

**Pull request merged**